src/apps/vpn/connectionbenchmark/benchmarktasktransfer.cpp

```diff
@@ -229,23 +229,13 @@ void BenchmarkTaskTransfer::transferProgressed(qint64 bytesSent,
   Q_UNUSED(bytesTotal);
 #endif
 
-  switch (m_type) {
-    case BenchmarkDownload: {
-      // Count and discard downloaded data
-      m_bytesTransferred += reply->skip(bytesTotal);
```

Comment:
So, now that we are connecting to both `QIODevice::readyRead()` and `QNetworkReply::downloadProgress()` we find ourselves with a bit of a race condition if both are attempting to manipulate the `QIODevice`. For example, if `readyRead()` happens first, then `QIODevice::skip()` will always report that no bytes have been skipped, because we already consumed them.

Luckily for us, for the job of counting the bytes transferred, we don't have to do anything special. The `QNetworkReply::downloadProgress()` already tells us the transfer via the signal arguments and we just need to save them somewhere.

But that still leaves the job of discarding incoming data during the benchmark so that we don't needlessly waste memory, and that's where the call to `NetworkReply::discardData()` comes in, which simply flushes the contents of `m_replyData`